main.tf

```diff
@@ -63,9 +82,26 @@ locals {
   existing_kms_instance_guid = var.kms_encryption_enabled == true ? element(split(":", var.existing_kms_instance_crn), length(split(":", var.existing_kms_instance_crn)) - 3) : null
 }
 
+# Create IAM Authorization Policies to allow event notification to access kms for the encryption key
+resource "ibm_iam_authorization_policy" "cos_policy" {
+  count                       = var.cos_integration_enabled == false || var.skip_en_cos_auth_policy ? 0 : 1
+  source_service_name         = "event-notifications"
+  source_resource_instance_id = ibm_resource_instance.en_instance.guid
+  target_service_name         = "cloud-object-storage"
+  target_resource_instance_id = var.cos_instance_id
+  roles                       = ["Reader"]
+  description                 = "To collect the events which failed delivery for your Event Notification instance ${ibm_resource_instance.en_instance.guid}."
```

Comment:
This should be a description of the auth policy. So suggest going with:

"Allow EN instance with GUID ${ibm_resource_instance.en_instance.guid} read access to the COS instance with ID ${var.cos_instance_id}"